curp/src/server/raw_curp/mod.rs

```diff
@@ -519,6 +530,51 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
         Ok((st_w.term, self_spec_pool))
     }
 
+    /// Handle `pre_vote`
+    /// Return `Ok(term, spec_pool)` if the vote is granted
+    /// Return `Err(term)` if the vote is rejected
+    pub(super) fn handle_pre_vote(
+        &self,
+        term: u64,
+        candidate_id: ServerId,
+        last_log_index: LogIndex,
+        last_log_term: u64,
+    ) -> Result<(u64, Vec<PoolEntry<C>>), u64> {
+        debug!(
+            "{} received vote: term({}), last_log_index({}), last_log_term({}), id({})",
+            self.id(),
+            term,
+            last_log_index,
+            last_log_term,
+            candidate_id
+        );
+
+        let st_r = self.st.read();
+        let log_r = self.log.read();
+
+        // calibrate term
+        if term < st_r.term {
+            return Err(st_r.term);
+        }
+        if term > st_r.term {
+            let timeout = st_r.follower_timeout_ticks;
+            if st_r.leader_id.is_some() && self.ctx.election_tick.load(Ordering::Acquire) < timeout
+            {
+                return Err(st_r.term);
+            }
+        }
+
+        // check if the candidate's log is up-to-date
+        if !log_r.log_up_to_date(last_log_term, last_log_index) {
+            return Err(st_r.term);
+        }
+
+        // grant the vote
+        debug!("{} pre votes for server {}", self.id(), candidate_id);
+        self.reset_election_tick();
```

Comment:
If we reset election tick here, will the follower accept the pre-vote but reject the vote?

Reply:
There really shouldn't be a reset here. Removed